### PR TITLE
Recipe App: Add foods integration spec

### PR DESCRIPTION
**Pull request by gtekle**
## In this pull request, I have:
- added system spec for `foods list` view
- added system spec for `create new food` view
- added system spec for user `Sign in` page

### CODE REVIEW

File: spec/system/foods/index_system_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    within('#new_user') do
+      fill_in 'Email', with: 'adam@domain.com'
+      fill_in 'Password', with: 'password'
+    end
+    click_button 'Log in'
+    visit foods_path
+
+    expect(current_path).to eq foods_path
+    expect(page).to have_content('Apple')
+  end
```
Comment (guerreiropedr0) [inline spec/system/foods/index_system_spec.rb:37]: I think we should do more integration specs here, what about testing if the Add Food link redirects you to another path and the delete button works?

File: spec/system/foods/new_system_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  it 'add new food should create new food' do
+    visit '/users/sign_in'
+    within('#new_user') do
+      fill_in 'Email', with: 'adam@domain.com'
+      fill_in 'Password', with: 'password'
+    end
+    click_button 'Log in'
+    visit foods_path
+
+    click_link('Add Food')
```
Comment (guerreiropedr0) [inline spec/system/foods/new_system_spec.rb:17]: This is a good test for the index view, but I think we should start from the new_food_path here, then test if filling in works or not.

You should also test what happens when the user doesn't fill every necessary field

**Review by guerreiropedr0 — CHANGES_REQUESTED** [review]
Hi @gtekle ,

You did extremely great on this project! 👏 👏
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 

## Good Points 👍
- Descriptive Pull Request
- Descriptive commit messages
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: spec/system/foods/index_system_spec.rb (hunk 1 above)
Comment (gtekle) [reply spec/system/foods/index_system_spec.rb:37]: Ok noted.
that was good catch.

File: spec/system/foods/new_system_spec.rb (hunk 2 above)
Comment (gtekle) [reply spec/system/foods/new_system_spec.rb:17]: good catch.

**Review by guerreiropedr0 — APPROVED** [review]
Hi @gtekle 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:


nodding-yes

## Optional suggestions
- Check the single comments :smiley:

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------

**Pull request merged**